internal/debug/flags.go

```diff
@@ -55,6 +55,11 @@ var (
 		Name:  "debug",
 		Usage: "Prepends log messages with call-site location (file and line number)",
 	}
+	swaplogpathFlag = cli.StringFlag{
+		Name:  "slogpath",
```

Comment:
The default is an empty String, so what happens if the flag is not provided? Do we not create the file at all? Did I miss this part?

Reply:
If the string is not provided, the RotatingFileHandler will not be created.
When the new logger tries to retrieve the handler it will default to the swarm logger output.

```
hd, ok := h.(log.Handler)

	if !ok {
		//default handler just in case the handler is not set
		hd = log.Root().GetHandler()
		swapLogger.SetHandler(hd)
		return swapLogger
	}
```
This so i dont change the default behaviuor in the Setup method.
I could also refactor and leave duplicate code, with this specific logic.

Reply:
I think in that case we should have a default log location. I wonder what core thinks about this though.

Reply:
If the path is not provided it will use the swarm logger as default.

Reply:
I understand. My doubt is whether it would be better if we have the file anyways. I understand your position to be "only if the user wants the file, let's create it". It makes sense, but I am wondering if we should have such a file by default in swarm's data directory somewhere.
@nonsense @janos @skylenet